lib/document_viewer_web/controllers/document_controller.ex

```diff
@@ -0,0 +1,21 @@
+defmodule DocumentViewerWeb.DocumentController do
+  use DocumentViewerWeb, :controller
+
+  def show(conn, %{"bucket_name" => bucket_name, "file_path" => file_path}) do
+    config_fn = Map.get(conn.assigns, :config_fn, &ExAws.Config.new/1)
+    presigned_url_fn = Map.get(conn.assigns, :presigned_url_fn, &ExAws.S3.presigned_url/4)
+    url = presigned_url(bucket_name, file_path, config_fn, presigned_url_fn)
+
+    redirect(conn, external: url)
```

Comment:
I know there was some concern from Lisa W about having people download the PDFs to read them during the discussions about the RFS vendor. Is that a concern here?

Reply:
My understanding is that downloading documents is a necessary function of this app—one of the primary use cases is providing documents in response to requests from legal. My working idea was to provide bare bones functionality the simplest way possible, and use that as a concrete starting point to use for thinking through improvements (quite likely for usability as well). Does that seem reasonable, or do you have any suggestions on the approach?

Reply:
I think this approach is fine, if the purpose of the application is for legal to download the documents. In that case though, you might want to have the application fetch the data from S3 and provide it directly, along with a `content-disposition: attachment; filename="XXX.pdf"` header so that the browser offers it as a download.

Reply:
This conversation is helpful. My initial reaction to your suggestion was that the user would want to know it was really the document they were looking for before downloading it. So at least in my imagined scenario there seem to 2 user stories: I want to preview documents to find the one I'm looking for, and I want to download a document once I know I have the right one.

Give that I expanded the interface. So now when you click on a result you get to a show page where you see the document embedded. If it looks correct you can use a Download button to explicitly download it. It seems like this is better both for the user experience, and in terms of logging discrete actions for security. What do you think?